src/hooks/useStyledSafeAreaInsets.ts

```diff
@@ -20,16 +22,27 @@ import useStyleUtils from './useStyleUtils';
  *     // Use these values to style your component accordingly
  * }
  */
-function useStyledSafeAreaInsets(safeAreaInsetsPercentage?: number) {
+function useStyledSafeAreaInsets() {
     const StyleUtils = useStyleUtils();
     const insets = useSafeAreaInsets();
+    const {paddingTop, paddingBottom} = StyleUtils.getSafeAreaPadding(insets);
+
+    const screenWrapperStatusContext = useContext(ScreenWrapperStatusContext);
+    const isSafeAreaTopPaddingApplied = screenWrapperStatusContext?.isSafeAreaTopPaddingApplied ?? false;
+    const isSafeAreaBottomPaddingApplied = screenWrapperStatusContext?.isSafeAreaBottomPaddingApplied ?? false;
+
+    const adaptedInsets = {
+        ...insets,
+        top: isSafeAreaTopPaddingApplied ? 0 : insets?.top,
+        bottom: isSafeAreaBottomPaddingApplied ? 0 : insets?.bottom,
+    };
+    const adaptedPaddingBottom = isSafeAreaBottomPaddingApplied ? 0 : paddingBottom;
 
-    const {paddingTop, paddingBottom} = StyleUtils.getSafeAreaPadding(insets, safeAreaInsetsPercentage);
     return {
-        paddingTop,
-        paddingBottom,
-        insets,
-        safeAreaPaddingBottomStyle: {paddingBottom},
+        paddingTop: isSafeAreaTopPaddingApplied ? 0 : paddingTop,
+        paddingBottom: adaptedPaddingBottom,
+        insets: adaptedInsets,
+        safeAreaPaddingBottomStyle: {adaptedPaddingBottom},
```

Comment:
If this is a style object shouldn't it be `{paddingBottom: adaptedPaddingBottom}`?